src/mm-input/mm-input.js

```diff
@@ -4,213 +4,175 @@
-
-		if (valid) {
-			this.error = false;
-		} else {
-			this.error = true;
-		}
-	},
-
-	validator: function() {
-		switch(this.validation) {
-			case "empty":
-				return this.value.length > 0;
-			case "int":
-				return this.inter.test(this.value);
-			case "decimal":
-				return this.floater.test(this.value);
-			case "email":
-				return this.email.test(this.value);
-			case "alpha":
-				return this.alpha.test(this.value);
-			default:
-				return true;
-		}
-	},
+		},
 
-	updateUI: function() {
-		var dataFilter = function(a) {
-			return a.name.indexOf(this.value) != -1;
-		};
-
-		//handle the clear and search state styling:
-		if (this.value && this.value.length > 0) {
-			if (this.clear) {
-				this.$.clearBtn.style.display = "block";
-			}
-			if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
-				this.$.icon.style.display = "none";
-			}
-			if (this.autocomplete) {
-				this.searchItems = this.data.filter(dataFilter.bind(this));
-				if (this.searchItems.length > 0) {
-					this.open();
-				}
-				// if (this.searchItems.length > this.maxItems) {
-				// this.searchItems = this.searchItems.splice(0, this.maxItems);
-				// }
+		_searchChanged: function(newVal, oldVal) {
+			if (newVal) { 
+				this.icon = "search";
+			} else if (!newVal && this.icon === "search") {
+				this.icon = false;
 			}
-		} else {
-			if (this.clear) {
-				this.$.clearBtn.style.display = "none";
-			}
-			if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
+		},
+
+		_iconChanged: function(newVal, oldVal) {
+			if (newVal) { 
 				this.$.icon.style.display = "block";
+			} else {
+				this.$.icon.style.display = "none";
 			}
-			if (this.autocomplete) {
-				this.close();
+		},
+		
+		_valueChanged: function(newVal, oldVal) {
+			if (newVal) {
+				this.updateUI(newVal);
+				this.fire("changed", { value: newVal });
 			}
-		}
-	},
-	
-	bindModel: function(model, property) {
-		this.model = model;
-		this.property = property;
-	},
+		},
 
-	updateModel: function() {
-		if (this.model && this.property) {
-			//check for BB models
-			if (this.model.set) {
-				var o = {};
-				o[this.property] = this.value;
-				this.model.set(o);
+		updateUI: function(value) {
+			// handles clear and search/icon visual state:
+			if (value) {
+				if (this.clear) {
+					this.$.clearBtn.style.display = "block";
+				}
+				if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
+					this.$.icon.style.display = "none";
+				}
 			} else {
-				this.model[this.property] = this.value;
+				if (this.clear) {
+					this.$.clearBtn.style.display = "none";
+				}
+				if (this.icon === "search" && this.search && this.clear || this.icon && this.clear) {
+					this.$.icon.style.display = "block";
+				}
 			}
+		},
+
+		clearInput: function(e) {
+			this.value = "";
+			this.updateUI(false);
+		},
+
+		updateStyle: function(icon, width, fitparent, clear) {
```

Comment:
This was intentionally part of the public api... don't remember the back story on that. It is never called internally, so it could be removed, but that would/could be a breaking change.